### PR TITLE
Use exit signals

### PR DESCRIPTION
**Pull request by josevalim**
For compatibility with GenStage master.

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
@josevalim mind bringing in the newest master here? With that, we can begin testing this w/ https://github.com/elixir-lang/gen_stage/pull/155 as well.

**Comment by josevalim** [comment]
@zeeshanlakhani done!

File: lib/flow/coordinator.ex [hunk 1]
```diff
@@ -66,9 +71,9 @@ defmodule Flow.Coordinator do
     {:noreply, state}
   end
 
-  def handle_info({:"$gen_producer", {consumer, ref}, {:subscribe, _, _}}, %{intermediary: intermediary} = state) do
+  def handle_info({:"$gen_producer", {consumer, ref}, {:subscribe, _, opts}}, %{intermediary: intermediary} = state) do
     for {pid, _} <- intermediary do
-      subscribe(consumer, pid)
+      GenStage.async_subscribe(consumer, [to: pid] ++ opts)
```
Comment (zeeshanlakhani) [inline lib/flow/coordinator.ex]: 👍

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by zeeshanlakhani** [comment]
@fishcakez and @josevalim great to see this merged-in as well, as we've been using this branch for a good bit. Yay :heart:.

**Pull request merged**